code/hsec-core/src/Security/Advisories/Core/Advisory.hs

```diff
@@ -44,10 +48,30 @@ data Advisory = Advisory
   }
   deriving stock (Show)
 
+data Ecosystem = Hackage Text | GHC GHCComponent
```

Comment:
This is more than the ecosystem: it is a component/package identifier qualified (via constructor) by ecosystem.  I approve of the structure of this data type, but I would prefer to name it something else.  Some suggestions:

- `PackageIdentifier`
- `AffectedPackageIdentifier`
- `ComponentIdentifier`
- `AffectedComponentIdentifier`

I lean toward `ComponentIdentifier` (the fact that the component is *affected* by some issue is incidental to its construction).  This is because, in future, we might further refine the type to allow specifying a sub-package *component* (e.g. `hsec-tools:exe`, if there were an issue in the executable but the `lib` component was unaffected).